### PR TITLE
Fix `DefaultRoleService` unintended case-sensitive handling of usernames

### PR DESCRIPTION
**Pull request by ghsolomon**

### CODE REVIEW

File: src/main/groovy/io/xh/hoist/role/provided/DefaultRoleService.groovy [hunk 1]
```diff
@@ -32,8 +32,8 @@ import static io.xh.hoist.util.InstanceConfigUtils.getInstanceConfig
  * Applications using this default implementation must either:
  *
  *  1) Define a `RoleService` class that extends this class. Allows for overriding its protected
- *     methods to customize behavior, including the option to implement `getUsersForDirectoryGroups`
- *     to resolve external directory group memberships (see below).
+ *     methods to customize behavior, including the option to implement
+ *     `doLoadUsersForDirectoryGroups` to resolve external directory group memberships (see below).
```
Comment (amcclain) [inline src/main/groovy/io/xh/hoist/role/provided/DefaultRoleService.groovy:36]: Thanks for catching!

**Review by amcclain — COMMENTED** [review]
Wondering if we want to take this further and have our `RoleMember` class normalize its `name` field to lowercase when `type == USER`.  We could do that using the GORM lifecycle hooks. It would get the data at rest into a more normal form and help ensure our GORM-level uniqueness constraint works as expected.

Likewise, we could leverage our `loadUsersForDirectoryGroups` wrapper around the app-supplied implementation method to lowercase all of the usernames emitted there, getting them into the lowercase format sooner so they show up that way (for better, I think) in the Admin UI.

Regardless, I think it's still the right choice to force to lowercase in the places you've added, to handle the small numbers of existing records that might already exist if nothing else.

**Comment by lbwexler** [comment]
@amcclain -- was thinking about the same, but wondering whether clients that wanted to so see their usernames in the admin screen according to their own corporate conventions might prefer we not force the lowercasing at that stage.

Agree this is a good change in any case, even if we want to take it further at some point, and I will merge in prep for release

**Comment by amcclain** [comment]
Got it, thanks.  I was a bit wary that some of the "effective" methods baked into `Role` could get thrown off by dupes with different casing, or something like that.  Not sure if that's a real issue

**Pull request merged**